File: mcp/src/stdio-server.ts
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
import { z } from "zod";

const server = new McpServer(
  { name: "aiox-local-mcp", version: "0.1.0" },
  {
    instructions:
      "Servidor MCP de exemplo neste repositório. Oferece ferramenta echo, recurso de metadados e um prompt simples para testar integrações (Cursor, Inspector, clientes MCP).",
  }
);

server.registerTool(
  "echo",
  {
    description: "Devolve o texto recebido. Útil para validar stdio e permissões do cliente.",
    inputSchema: { message: z.string().describe("Texto a devolver") },
  },
  async ({ message }) => ({
    content: [{ type: "text", text: message }],
  })
);

server.registerTool(
  "repo_ping",
  {
    description: "Confirma que o servidor MCP local está ativo e informa o repositório.",
    inputSchema: {},
  },
  async () => ({
    content: [
      {
        type: "text",
        text: "MCP local aiox-squads: servidor stdio OK (TypeScript SDK @modelcontextprotocol/sdk).",
      },
    ],
  })
);

server.registerResource(
  "about_mcp_local",
  "workspace://mcp/about",
  {
    title: "Sobre este MCP",
    description: "Metadados do servidor MCP empacotado em mcp/",
    mimeType: "text/markdown",
  },
  async () => ({
    contents: [
      {
        uri: "workspace://mcp/about",
        mimeType: "text/markdown",
        text: [
          "# MCP local (aiox-squads)",
          "",
          "- Pasta: `mcp/`",
          "- SDK: [@modelcontextprotocol/sdk](https://github.com/modelcontextprotocol/typescript-sdk)",
          "- Inspector: `npm run inspector` (UI) ou `npm run inspector:config`",
          "- Docs: https://modelcontextprotocol.io",
        ].join("\n"),
      },
    ],
  })
);

server.registerPrompt(
  "summarize_intent",
  {
    title: "Resumir intenção",
    description: "Template mínimo para testar listagem de prompts no cliente.",
    argsSchema: { topic: z.string().describe("Assunto a resumir") },
  },
  ({ topic }) => ({
    messages: [
      {
        role: "user",
        content: {
          type: "text",
          text: `Em uma frase, qual é a intenção principal em relação a: ${topic}?`,
        },
      },
    ],
  })
);

const transport = new StdioServerTransport();
await server.connect(transport);

File: mcp/python/stdio_server.py
"""
Servidor MCP mínimo com o SDK Python official (FastMCP).

Repositório de referência: https://github.com/modelcontextprotocol/python-sdk

Servidor stdio (fica à espera do cliente; use com Cursor ou outro host MCP):

    npm run server:py

    # ou, com venv ativo:
    .venv\\Scripts\\mcp run python/stdio_server.py -t stdio

Teste rápido (Inspector em CLI, na pasta mcp/ — arranca o servidor, lista tools e chama echo/ping):

    npm run test:py

UI do Inspector: npm run inspector:python
"""

from mcp.server.fastmcp import FastMCP

mcp = FastMCP(
    "aiox-python-mcp",
    instructions="Servidor MCP de exemplo em Python (FastMCP) no monorepo aiox-squads.",
)


@mcp.tool()
def echo_py(message: str) -> str:
    """Devolve o texto (teste do SDK Python)."""
    return message


@mcp.tool()
def ping_py() -> str:
    """Confirma que o servidor Python está ativo."""
    return "MCP Python (mcp package) OK."


if __name__ == "__main__":
    mcp.run(transport="stdio")
